config/700-websocket-envoyfilter.yaml

```diff
@@ -0,0 +1,42 @@
+# Copyright 2022 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This Envoyfilter is needed to support websocket for DomainMapping.
+apiVersion: networking.istio.io/v1alpha3
+kind: EnvoyFilter
+metadata:
+  name: dm-websocket-filter
+  namespace: istio-system
+  labels:
+    app.kubernetes.io/component: net-istio
+    app.kubernetes.io/name: knative-serving
+spec:
+  workloadSelector:
+    labels:
+      app: istio-ingressgateway
```

Comment:
I think this will not work when we use another ingress-gateway instead of the default `istio-ingressgateway`.
Do you have any good idea how we can take care of it? Or we will just document it?

Reply:
That's true. Currently we already hard coded the selector though in our Gateway https://github.com/knative-sandbox/net-istio/blob/main/config/203-local-gateway.yaml#L31 and https://github.com/knative-sandbox/net-istio/blob/main/config/202-gateway.yaml#L29. I am curious what mechanism we use for those Gateways?